tasks/main.yml

```diff
@@ -23,6 +23,17 @@
     state: present
   when:
     - not network_packages is subset(ansible_facts.packages.keys())
+  register: package_install
+
+# If network packages have changed and wireless connections are required,
+# NetworkManager must be restarted
+- name: Restart NetworkManager
+  service:
+    name: NetworkManager
+    state: restarted
+  when:
+    - wireless_connections_defined
+    - package_install.changed  # noqa 503
```

Comment:
This is for Ansible lint https://docs.ansible.com/ansible-lint/rules/rules.html#excluding-rules. It complains that we should use a handler instead of restarting the service if something is changed. A handler is not appropriate in this case as NM needs to be restarted prior to creating the wireless interfaces

Reply:
How about adding a link to https://docs.ansible.com/ansible-lint/rules/default_rules.html to make this is easier to check in case we keep the task.

Reply:
Have added the link now